File: source-connector/src/test/java/sourceconnector/domain/processor/impl/EmptyFilterProcessorTest.java
package sourceconnector.domain.processor.impl;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import sourceconnector.domain.log.JSONLog;
import sourceconnector.domain.log.Log;
import sourceconnector.domain.log.LogMetadata;
import sourceconnector.domain.processor.impl.EmptyFilterProcessor;

import static org.assertj.core.api.Assertions.assertThat;

class EmptyFilterProcessorTest {

  @DisplayName("Should result is to be null when payload is empty")
  @Test
  void shouldReturnFalseForEmptyInput() {
    // given
    EmptyFilterProcessor processor = new EmptyFilterProcessor();
    Log emptyInput = new JSONLog("", LogMetadata.EMPTY);

    // when
    Log result = processor.process(emptyInput);

    // then
    assertThat(result).isNull();
  }

  @DisplayName("Should result is to be null when payload is blank")
  @Test
  void shouldReturnFalseForBlankInput() {
    // given
    EmptyFilterProcessor processor = new EmptyFilterProcessor();
    Log emptyInput = new JSONLog("  ", LogMetadata.EMPTY);

    // when
    Log result = processor.process(emptyInput);

    // then
    assertThat(result).isNull();
  }
}

File: source-connector/src/test/kotlin/sourceconnector/domain/processor/impl/EmptyFilterProcessorTest.kt
package sourceconnector.domain.processor.impl

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import sourceconnector.domain.log.JSONLog
import sourceconnector.domain.log.Log
import sourceconnector.domain.log.LogMetadata

internal class EmptyFilterProcessorTest {
  @DisplayName("Should result is to be null when payload is empty")
  @Test
  fun shouldReturnFalseForEmptyInput() {
    // given
    val processor = EmptyFilterProcessor()
    val emptyInput: Log = JSONLog("", LogMetadata.EMPTY)

    // when
    val result = processor.process(emptyInput)

    // then
    assertThat<Log?>(result).isNull()
  }

  @DisplayName("Should result is to be null when payload is blank")
  @Test
  fun shouldReturnFalseForBlankInput() {
    // given
    val processor = EmptyFilterProcessor()
    val emptyInput: Log = JSONLog("  ", LogMetadata.EMPTY)

    // when
    val result = processor.process(emptyInput)

    // then
    assertThat<Log?>(result).isNull()
  }
}
